rr_iarrc/src/finish_line_watcher/finish_line_watcher.cpp

```diff
@@ -78,23 +74,85 @@ void ImageCB(const sensor_msgs::ImageConstPtr& msg) {
     frame = cv_ptr->image;
     blockEnvironment(frame);
 
-    auto count = countNonZero(frame);
+    // Do contour detection
+    vector<vector<Point>> contours;
+    vector<Vec4i> hierarchy;
+    Mat contourImg = frame.clone();
+    findContours(contourImg, contours, hierarchy, CV_RETR_LIST, CV_CHAIN_APPROX_SIMPLE);
+
+    Mat drawing = Mat::zeros(contourImg.size(), CV_8UC3);
+    double maxLength = 0.0;
+    double maxLengthYPos = 0.0;
+    for (size_t i = 0; i < contours.size(); i++) {
+        // Find the average slopes of the contour
+        Point start = contours.at(i).at(0);
+        Point end = contours.at(i).at((contours.at(i).size() - 1) / 2);
```

Comment:
My intention of how this would work is the following:

- Look at the HSV thresholded image for the bright pink
- Detect _lines_ in the image
- Filter the lines as to get rid of those that are quite short or those that have a high slope. Looking at the bag files it looks like when looking for the finish line the car is coming in at more or less straight on, so we don't want to include any lines that are particularly vertical, thus adding the slope restriction.
- If there are still lines left after this filtering then I assume that's the finish line.

I now realize that I was looking at contours, not lines, which is why calculating slope seems fairly nonsensical. I will update to use line detection, not contour detection. It seemed that sometimes contour detection would split the finish line up into two contours, which is why I didn't want to use it.